guides/source/active_record_migrations.md

```diff
@@ -579,6 +579,33 @@ remove_foreign_key :accounts, :branches
 remove_foreign_key :accounts, column: :owner_id
 ```
 
+### Check Constraints
+
+While it's not required you might want to add check constraints to limit the
+value range that can be placed in a column.
+
+```ruby
+add_check_constraint :products, "price > 0", name: "price_check"
+```
+
+This [`add_check_constraint`][] call adds a new check constraint to the
+`products` table.
+The check constraint guarantees that a value in the `price` column in the
+`products` table falls within the value range set by the `expression`.
+
+`add_check_constraint` also supports options such as `name`, and `validate`
+(PostgreSQL only).
+
+Check constraints can also be removed:
+
+```ruby
+remove_check_constraint :products, name: "price_check"
+```
+
+The `expression` parameter will be ignored if present. It can be helpful to
+provide this in a migration's change method so it can be reverted. In that case,
+ expression will be used by `add_check_constraint`.
```

Comment:
I would like to put most of this information in the API docs instead (i.e. add whatever is missing to [here](https://github.com/rails/rails/blob/d6a41ee4191a7031444152231778f38b4089597e/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb#L1162-L1176) and [here](https://github.com/rails/rails/blob/d6a41ee4191a7031444152231778f38b4089597e/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb#L1192-L1199)), and then link to them as you have done below.

The benefit of putting the information in the API docs is that they are more likely to be kept up to date when there are code changes, and they are more discoverable thanks to RDoc-related tooling.  Conversely, putting the information in the guide can increase the maintenance burden by adding a 2nd place where documentation must be updated.

Also, guides have a natural tendency to accumulate information over time, and core points may be drowned out by niche information.  I think it is better for the API docs to provide detailed information and limited-scope usage examples, while the guides provide higher-level guidance and cross-cutting usage examples.